### PR TITLE
Problem: (CRO-678) Tendermint HttpRpcClient compilation error

### PR DESCRIPTION
**Pull request by calvinlauco**
Solution: Fix compilation error

---
- Add `cargo build` test in TravisCI
- Port changes from `WebSocketRpcClient` to `HttpRpcClient` to fix compilation error

### CODE REVIEW

**Comment by calvinlauco** [comment]
TravisCI fails when doing `cargo audit` because of missing `Cargo.lock` in https://github.com/crypto-com/tendermint-rs
https://travis-ci.org/crypto-com/chain/jobs/630071998#L1214

The straightforward fix would be to add one. But I believe it is intended not to provide one. @yihuang  Actually this does not occur in other PR like:
https://github.com/crypto-com/chain/pull/729
https://travis-ci.org/crypto-com/chain/jobs/630094951#L1682

@devashishdxt @yihuang Do you know why this would happen?

File: chain-abci/src/app/commit.rs [hunk 1]
```diff
@@ -172,13 +172,13 @@ impl<T: EnclaveProxy> ChainNodeApp<T> {
...
         let encoded_height = i64::encode_var_vec(new_state.last_block_height);
         inittx.put(COL_APP_HASHS, &encoded_height, &new_state.last_apphash);
-        if self.tx_query_address.is_some() {
-            inittx.put(
-                COL_APP_STATES,
-                &encoded_height,
-                &new_state.top_level.encode(),
-            );
-        }
+        // if self.tx_query_address.is_some() {
```
[the remark this answers is not in the document]
Comment (devashishdxt) [reply chain-abci/src/app/commit.rs]: @calvinaco  Is this change related to the problem mentioned in PR title?

[the remark this answers is not in the document]
Comment (calvinlauco) [reply chain-abci/src/app/commit.rs]: No it is not intended, I will push a new commit to revert it.

**Comment by devashishdxt** [comment]
> TravisCI fails when doing `cargo audit` because of missing `Cargo.lock` in https://github.com/crypto-com/tendermint-rs
> https://travis-ci.org/crypto-com/chain/jobs/630071998#L1214
> 
> The straightforward fix would be to add one. But I believe it is intended not to provide one. @yihuang Actually this does not occur in other PR like:
> #729
> https://travis-ci.org/crypto-com/chain/jobs/630094951#L1682
> 
> @devashishdxt @yihuang Do you know why this would happen?

I don't know the reason for this. But, adding `Cargo.lock` to `tendermin-rs` should not be the solution here. Can you try merging latest master?

**Comment by yihuang** [comment]
> TravisCI fails when doing `cargo audit` because of missing `Cargo.lock` in https://github.com/crypto-com/tendermint-rs
> https://travis-ci.org/crypto-com/chain/jobs/630071998#L1214
> 
> The straightforward fix would be to add one. But I believe it is intended not to provide one. @yihuang Actually this does not occur in other PR like:
> #729
> https://travis-ci.org/crypto-com/chain/jobs/630094951#L1682
> 
> @devashishdxt @yihuang Do you know why this would happen?

Have no idea, most libraries don't provide `Cargo.lock` file.

**Comment by tomtau** [comment]
that failure with Cargo.lock is strange, as other libraries (e.g. https://github.com/crypto-com/rust-secp256k1-zkp) don't have it either and it's fine.

**Pull request merged**